[WIP] Add unicode support for linebreak and textwidth

worldeva opened the pull request:
Closes https://github.com/neovim/neovim/issues/13967

- [x] Implement a function to check the block of a Unicode Codepoint
- [ ] Implement tests for `unicode_block`
- [ ] Add option to set what blocks are linebroken on
- [ ] Make `vim_isbreak` take multibyte characters and check the block of them
- [ ] Make `internal_format` check block of characters

worldeva committed:
mbyte.h: Add UnicodeBlock enum

worldeva committed:
mbyte.c: Create unicode_block_table

worldeva committed:
mbyte.c: Add utf_block function, fix const before static warning

`static` moved to before `const`

bfredl commented on src/nvim/mbyte.c:
```diff
@@ -1054,6 +1369,31 @@ bool utf_printable(int c)
+{
+  int bottom = 0;
+  int top = ARRAY_SIZE(unicode_block_table) - 1;
+  int middle = 0; //Start at zero so that latin chars are checked quickly
```
hmm I think it could be clearer to spell this out as an extra `if()` statement in the beginning of the function (which how it typically is done).

worldeva committed:
screen.c: Modify `win_line` to pass multibyte character to vim_isbreak

You know the pain when you spend a few days trying to fix a bug that
doesn't exist? Yeah, that was me. Note: `ptr` seems to have a "ghost" value past the end of the line.

worldeva committed:
charset.c: Modify `win_lbr_chartabsize` to pass multibyte character to vim_isbreak

worldeva commented:
I... I got half way through before realizing that a better method of implementation could be to have users specify the unicode block of a character (`3040..309F,...`), since they'd have to look up the block names anyways. Thoughts?
